packages/firestore/src/model/mutation.ts

```diff
@@ -385,26 +441,27 @@ export function mutationEquals(left: Mutation, right: Mutation): boolean {
   }
 
   if (left.type === MutationType.Set) {
-    return (left as SetMutation).value.isEqual((right as SetMutation).value);
+    const setLeft: SetMutation = left as SetMutation;
+    const setRight: SetMutation = right as SetMutation;
+    return (
+      setLeft.value.isEqual(setRight.value) &&
+      fieldTransformsAreEqual(setLeft.fieldTransforms, setRight.fieldTransforms)
```

Comment:
Can we move `fieldTransforms` up to the Mutation base class? I think the PR will become a lot simpler then. You can then check `fieldTransformsAreEqual` regardless of type and `MutationWithTransforms` might even go away.